src/CallObjectLoader.js

```diff
@@ -0,0 +1,228 @@
+    this._currentLoad && this._currentLoad.cancel();
+
+    // Start a new load
+    this._currentLoad = new LoadOperation(
+      meetingOrBaseUrl,
+      callFrameId,
+      () => {
+        successCallback(false); // false = "this load() wasn't a no-op"
+      },
+      failureCallback
+    );
+    this._currentLoad.start();
+  }
+
+  /**
+   * Cancel loading the call object bundle. No callbacks will be invoked.
+   */
+  cancel() {
+    this._currentLoad && this._currentLoad.cancel();
+  }
+
+  /**
+   * Returns a boolean indicating whether the call object bundle has been
+   * loaded and executed.
+   */
+  get loaded() {
+    return this._currentLoad && this._currentLoad.succeeded;
+  }
+}
+
+const LOAD_ATTEMPTS = 3;
+const LOAD_ATTEMPT_DELAY = 3 * 1000;
+
+class LoadOperation {
+  // Here failureCallback takes the same parameters as CallObjectLoader.load,
+  // and successCallback takes no parameters.
+  constructor(meetingOrBaseUrl, callFrameId, successCallback, failureCallback) {
+    this._attemptsRemaining = LOAD_ATTEMPTS;
+    this._currentAttempt = null;
+
+    this._meetingOrBaseUrl = meetingOrBaseUrl;
+    this._callFrameId = callFrameId;
+    this._successCallback = successCallback;
+    this._failureCallback = failureCallback;
+  }
+
+  start() {
+    // Bail if this load has already started
+    if (this._currentAttempt) {
+      return;
+    }
+
+    // console.log("[LoadOperation] starting...");
+
+    const retryOrFailureCallback = (errorMessage) => {
+      if (this._currentAttempt.cancelled) {
+        // console.log("[LoadOperation] cancelled");
+        return;
+      }
+
+      this._attemptsRemaining--;
+      this._failureCallback(errorMessage, this._attemptsRemaining > 0); // true = "will retry"
+      if (this._attemptsRemaining <= 0) {
+        // Should never be <0, but just being extra careful here
+        // console.log("[LoadOperation] ran out of attempts");
+        return;
+      }
+
+      setTimeout(() => {
+        if (this._currentAttempt.cancelled) {
+          // console.log("[LoadOperation] cancelled");
+          return;
+        }
+        this._currentAttempt = new LoadAttempt(
+          this._meetingOrBaseUrl,
+          this._callFrameId,
+          this._successCallback,
+          retryOrFailureCallback
+        );
+        this._currentAttempt.start();
+      }, LOAD_ATTEMPT_DELAY);
+    };
+
+    this._currentAttempt = new LoadAttempt(
+      this._meetingOrBaseUrl,
+      this._callFrameId,
+      this._successCallback,
+      retryOrFailureCallback
+    );
+    this._currentAttempt.start();
+  }
+
+  cancel() {
+    this._currentAttempt && this._currentAttempt.cancel();
+  }
+
+  get cancelled() {
+    return this._currentAttempt && this._currentAttempt.cancelled;
+  }
+
+  get succeeded() {
+    return this._currentAttempt && this._currentAttempt.succeeded;
+  }
+}
+
+class LoadAttemptAbortedError extends Error {}
+
+const LOAD_ATTEMPT_TIMEOUT = 20 * 1000;
+
+class LoadAttempt {
```

Comment:
Just to explain why I broke down the process of loading a bundle into discrete object instances (`LoadAttempt`s) like this, rather than maintaining state some other way: since each "attempt" can live beyond the lifetime of the `load()` call (since a `load()` may be cancelled while attempts are still ongoing), letting them live on in memory as objects that can clean themselves up was the natural model.